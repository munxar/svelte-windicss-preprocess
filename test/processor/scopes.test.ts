import { preprocess } from '../../src/index';
import { testConfig } from '../helpers/utils';

describe('Scopes test suite', () => {
  let result: string;
  let content = `
<div class="scoped bg-red-200">Hello, World!</div>
<style lang="postcss" scoped>
.scoped {
  @apply text-red-500;
}

@screen sm {
  .scoped {
    @apply text-blue-500;
  }
}
</style>
  `;

  it('should generate scoped classes locally', async () => {
    result = '';
    result = (
      await preprocess({ ...testConfig, compile: false }).markup({
        content,
        filename: 'test.svelte',
      })
    ).code;
    expect(result).toMatchSnapshot('localScopedClasses');
  });

  it('should generate scoped classes locally and compile', async () => {
    result = '';
    result = (
      await preprocess({ ...testConfig, compile: true }).markup({
        content,
        filename: 'test.svelte',
      })
    ).code;
    expect(result).toMatchSnapshot('localScopedClassesCompiled');
  });
});
