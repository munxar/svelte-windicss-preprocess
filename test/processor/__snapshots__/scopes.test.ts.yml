Templates test suite / should generate scoped classes locally / localScopedClasses / 0: |-

  <div class="scoped bg-red-200">Hello, World!</div>

  <style>
  :global(*), :global(::before), :global(::after) {
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    border-width: 0;
    border-style: solid;
    border-color: #e5e7eb;
  }
  :global(*) {
    --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgba(59, 130, 246, 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
  }
  :global(:root) {
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  }
  :global(:-moz-focusring) {
    outline: 1px dotted ButtonText;
  }
  :global(:-moz-ui-invalid) {
    box-shadow: none;
  }
  :global(::moz-focus-inner) {
    border-style: none;
    padding: 0;
  }
  :global(::-webkit-inner-spin-button), :global(::-webkit-outer-spin-button) {
    height: auto;
  }
  :global(::-webkit-search-decoration) {
    -webkit-appearance: none;
  }
  :global(::-webkit-file-upload-button) {
    -webkit-appearance: button;
    font: inherit;
  }
  :global([type='search']) {
    -webkit-appearance: textfield;
    outline-offset: -2px;
  }
  :global(abbr[title]) {
    -webkit-text-decoration: underline dotted;
    text-decoration: underline dotted;
  }
  :global(body) {
    margin: 0;
    font-family: inherit;
    line-height: inherit;
  }
  :global(html) {
    -webkit-text-size-adjust: 100%;
    font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
    line-height: 1.5;
  }
  .bg-red-200 {
    --tw-bg-opacity: 1;
    background-color: rgba(254, 202, 202, var(--tw-bg-opacity));
  }
  .scoped {
    --tw-text-opacity: 1;
    color: rgba(239, 68, 68, var(--tw-text-opacity));
  }
  @media (min-width: 640px) {
    .scoped {
      --tw-text-opacity: 1;
      color: rgba(59, 130, 246, var(--tw-text-opacity));
    }
  }
  </style>
Templates test suite / should generate scoped classes locally and compile / localScopedClassesCompiled / 0: |-

  <div class="windi-wca1md scoped">Hello, World!</div>

  <style>
  :global(*), :global(::before), :global(::after) {
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    border-width: 0;
    border-style: solid;
    border-color: #e5e7eb;
  }
  :global(*) {
    --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgba(59, 130, 246, 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
  }
  :global(:root) {
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  }
  :global(:-moz-focusring) {
    outline: 1px dotted ButtonText;
  }
  :global(:-moz-ui-invalid) {
    box-shadow: none;
  }
  :global(::moz-focus-inner) {
    border-style: none;
    padding: 0;
  }
  :global(::-webkit-inner-spin-button), :global(::-webkit-outer-spin-button) {
    height: auto;
  }
  :global(::-webkit-search-decoration) {
    -webkit-appearance: none;
  }
  :global(::-webkit-file-upload-button) {
    -webkit-appearance: button;
    font: inherit;
  }
  :global([type='search']) {
    -webkit-appearance: textfield;
    outline-offset: -2px;
  }
  :global(abbr[title]) {
    -webkit-text-decoration: underline dotted;
    text-decoration: underline dotted;
  }
  :global(body) {
    margin: 0;
    font-family: inherit;
    line-height: inherit;
  }
  :global(html) {
    -webkit-text-size-adjust: 100%;
    font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
    line-height: 1.5;
  }
  .scoped {
    --tw-text-opacity: 1;
    color: rgba(239, 68, 68, var(--tw-text-opacity));
  }
  .windi-wca1md {
    --tw-bg-opacity: 1;
    background-color: rgba(254, 202, 202, var(--tw-bg-opacity));
  }
  @media (min-width: 640px) {
    .scoped {
      --tw-text-opacity: 1;
      color: rgba(59, 130, 246, var(--tw-text-opacity));
    }
  }
  </style>
